library/ccm.c

```diff
@@ -144,9 +154,30 @@ void mbedtls_ccm_free( mbedtls_ccm_context *ctx )
         }                                                               \
                                                                         \
         for( i = 0; i < (len); i++ )                                    \
+        {                                                               \
             (dst)[i] = (src)[i] ^ b[i];                                 \
+            flow_ctrl_local++;                                          \
+        }                                                               \
     } while( 0 )
 
+/*
+ * Check flow_ctrl_local variable against a given length and reset a
+ * given flag if the flow is as expected. In other case, return
+ * MBEDTLS_ERR_PLATFORM_FAULT_DETECTED.
+ */
+#define CHECK_FLOW_CTRL( len, flow_flag )                               \
+    do                                                                  \
+    {                                                                   \
+        if( flow_ctrl_local == len )                                    \
```

Comment:
I'm sorry, but I cannot reply to your comment about UPDATE_CBC_MAC and CTR_CRYPT since the conversation is resolved.
The thing is that both of these macros use a lot of local variables - y, b, ctx, olen, ctr... Adding all of them as a parameter would decrease readability. But sure, I can add the flow ctrl_local parameter to `CHECK_FLOW_CTRL` .

Reply:
@AndrzejKurek , I moved the conversation to "Unresolved". I didn't see Piotr's question when marked it as resolved, sorry

Reply:
@shelib01 - no problem :) 
I have now added some readability improvements, as requested.